src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -377,23 +412,32 @@ private static string ResolvePath(string filename)
             }
 
             // Then check each directory listed in the PATH environment variables
+            return FindProgramInPath(filename);
+        }
+
+        /// <summary>
+        /// Gets the path to the program
+        /// </summary>
+        /// <param name="program"></param>
+        /// <returns></returns>
+        private static string FindProgramInPath(string program)
```

Comment:
The logic of this method is just moved from here as-is: 

https://github.com/dotnet/corefx/blob/fb0a5a037989d120cc4f42371eb1274cfc291442/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L380-L393